### PR TITLE
[READY] Sn32 develop redragon k630 bringup

### PR DESCRIPTION
**Pull request by bausensi**
## Description

Add a new data-driven keyboard support folder for Redragon K630 (Dragonborn).
Data-driven structure for RGB matrices isn't completely supported in mainline QMK, so
an accompanying config.h file with specific information has been supplied.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/redragon/k630/keymaps/via/config.h [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+/* Enable VIA RGB matrix and configure EEPROM emulation for VIA */
+#define VIA_QMK_RGBLIGHT_ENABLE
+#define WEAR_LEVELING_SN32_EMULATION_PAGE_COUNT 30
```
Comment (dexter93) [inline keyboards/redragon/k630/keymaps/via/config.h]:
```suggestion
```
this only needs to be overwritten from the default value, if the default value is deemed too small when building VIA. That is only the case on big keyboards, please verify if you need that

Comment (dexter93) [inline keyboards/redragon/k630/keymaps/via/config.h]:
```suggestion
```
this only needs to be overwritten from the default value, if the default value is deemed too small when building VIA. That is only the case on big keyboards, please verify if you need that

File: keyboards/redragon/k630/info.json [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+    "debounce": 3,
+    "diode_direction": "COL2ROW",
+    "matrix_pins": {
+        "cols": [ "A0",  "A1",  "A2",  "A3", "A4", "A5",  "A6",  "A7",  "A8", "A9", "A10", "A11", "A12", "A13" ],
+        "rows": [ "D11", "D10", "D9",  "D8", "D7" ]
+    },
+    
+    "tapping": {
+        "term": 150
+    },
```
Comment (dexter93) [inline keyboards/redragon/k630/info.json]: this should be needed when using tap dancing, but not a simple `LT` as you seem to be using in the default keymap ( afaik). If this is needed for a custom user keymap, you can nest a `info.json` with it in the keymap dir

File: keyboards/redragon/k630/info.json [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+        "caps_lock": "B14",
+        "on_state": 0
+    },
+    
+    "rgb_matrix": {
+        "driver": "SN32F24xB",
+        "hue_steps": 8,
+        "sat_steps": 8,
+        "val_steps": 8,
+        "speed_steps": 8,
```
Comment (dexter93) [inline keyboards/redragon/k630/info.json]: these seem to be here for convenience and not a hardware reason. The shared matrix driver supports 8/16/16/16 as default which is higher resolution. Probably best to limit those to userspace/keymap territory if you want convienience. As a rule of thumb, we want the keyboards in here to be as close to stock behavior as possible in the main configuration.

File: keyboards/redragon/k630/info.json [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "extrakey": true,
+        "mousekey": false,
+        "nkro": true,
+        "rgb_matrix": true
+    },
+
+    "debounce": 3,
```
Comment (dexter93) [inline keyboards/redragon/k630/info.json:28]: depending on your switches, you might be able to go as low as `1` here without compromising stability. A matter of preference I guess

**Review by dexter93 — COMMENTED** [review]
Overall pretty good and clean work. Thank you for the contribution. I left some nitpicky comments, but the bulk of it is good. Nice work

File: keyboards/redragon/k630/info.json (hunk 4 above)
Comment (bausensi) [reply keyboards/redragon/k630/info.json:28]: Understood. Making the changes right now.

**Comment by bausensi** [comment]
This PR is now finished. I will no longer make any major changes unless needed for merging from now on.

**Pull request merged**